aopy/analysis.py

```diff
@@ -202,6 +202,74 @@ def calc_success_rate(events, start_events=[b"TARGET_ON"], end_events=[b"REWARD"
     success_rate = n_success / n_trials
     return success_rate
 
+def calc_running_event_rate(event_name, event_record, event_timestamps, 
+                       window_size:float = 60., window_step = None):
+    
+    """
+    this function calculates running averages of event rates for each window with window_size stepped over window_step.
+    
+    Args:
+        event_name(string or int): event name e.g. b'REWARD' or 48
+        event_record(list, tuple, np.ndarray): event record with length of number of events
+        event_timstampts((list, tuple, np.ndarray)): timestamps in seconds with length of number of events
+        window_size(float): the running window size, defaults to 60 s or 1 minute
+        window_step(float): the step between betweens, if none, calculate the event rate with NON-OVERLAPPING windows
+        
+    Returns
+        a tuple of: 
+        runnning_event_rate(np.ndarray): an array with running event rates
+        window_timestamps(np.ndarray): timestamps at the middle of windows
```

Comment:
see some of ryan's code for exapmles how to properly format this. e.g. https://github.com/aolabNeuro/analyze/pull/151/files 

for example, don't use np.ndarray, instead use the size of the variables